src/app/routes.tsx

```diff
@@ -1,67 +1,115 @@
-import React, {Suspense, lazy} from 'react';
-import { createBrowserRouter, RouterProvider } from 'react-router-dom';
-import AppLayout from 'widgets/AppLayout';
+import React, { Suspense, lazy, ReactNode } from 'react';
+import { createBrowserRouter, RouterProvider, Navigate, Outlet, useLocation } from 'react-router-dom';
 import ErrorBoundary from 'shared/model/ErrorBoundary';
-import {Box, CircularProgress} from "@mui/material";
+import { Box, CircularProgress } from '@mui/material';
+import { useAppSelector } from 'shared/hooks/hooks';
+import AppLayout from 'widgets/AppLayout';
+
+const HomePage = lazy(() => import('pages/Home/ui/HomePage'));
+const SignInPage = lazy(() => import('pages/SignIn/ui/SignInPage'));
+const SignUpPage = lazy(() => import('pages/SignUp/ui/SignUpPage'));
+const ImageFeedPage = lazy(() => import('pages/ImageFeed/ui/ImageFeedPage'));
+const ImageEditorPage = lazy(() => import('pages/ImageEditor/ui/ImageEditorPage'));
+const WorkPage = lazy(() => import('pages/Work/ui/WorkPage'));
 
-const HomePage = lazy(()=>import('pages/Home/ui/HomePage'));
-const SignInPage = lazy(()=>import('pages/SignIn/ui/SignInPage'));
-const SignUpPage = lazy(()=>import('pages/SignUp/ui/SignUpPage'));
-const ImageFeedPage = lazy(()=>import('pages/ImageFeed/ui/ImageFeedPage'));
-const ImageEditorPage = lazy(()=>import('pages/ImageEditor/ui/ImageEditorPage'));
+interface RequireAuthProps {
+    children: ReactNode;
+}
+
+const RequireAuth: React.FC<RequireAuthProps> = ({ children }) => {
+    const auth = useAppSelector((state) => state.auth);
+    const location = useLocation();
+
+    if (!auth.userId) {
+        return <Navigate to="/signin" state={{ from: location }} replace />;
+    }
+
+    return <>{children}</>;
+};
+
+const PublicLayout: React.FC = () => {
+    const auth = useAppSelector((state) => state.auth);
+
+    return auth.userId ? <Navigate to="/feed" replace /> : (
+        <AppLayout>
+            <Outlet />
+        </AppLayout>
+    );
```

Comment:
```suggestion
if (auth.userId) {
   return <Navigate to="/feed" replace />
}

    return (
        <AppLayout>
            <Outlet />
        </AppLayout>
    );
```
please simplify conditions, avoid using ternary operators

Reply:
Done